benchmarks/nemo/nemolite2d/manual_versions/psykal_cpp_ompss/time_step_ompss.cpp

```diff
@@ -0,0 +1,274 @@
+#include <iostream>
+//#include <chrono>
+#include <vector>
+#include <cstdlib>
+
+// Kernels
+#include "../../kernels/c_family/continuity_kern.c"
+#include "../../kernels/c_family/momentum_u_kern.c"
+#include "../../kernels/c_family/momentum_v_kern.c"
+#include "../../kernels/c_family/boundary_conditions_kern.c"
+#include "../../kernels/c_family/time_update_kern.c"
+
+extern "C" void c_invoke_time_step(
+        // Fields
+        double * ssha_t,
+        double * sshn_t,
+        double * sshn_u,
+        double * sshn_v,
+        double * hu,
+        double * hv,
+        double * un,
+        double * vn,
+        double * ua,
+        double * ht,
+        double * ssha_u,
+        double * va,
+        double * ssha_v,
+        // Grid
+        int * tmask,
+        double * area_t,
+        double * area_u,
+        double * area_v,
+        double * dx_u,
+        double * dx_v,
+        double * dx_t,
+        double * dy_u,
+        double * dy_v,
+        double * dy_t,
+        double * gphiu,
+        double * gphiv,
+        // Scalars
+        int istep,
+        int internal_xstart,
+        int internal_xstop,
+        int internal_ystart,
+        int internal_ystop,
+        int width,
+        double rdt,
+        double cbfr,
+        double visc,
+        double omega,
+        double d2r,
+        double g
+        ){
+
+#define TASK_SIZE 32
+//Each double loop over jj then ji is divided into a new structure.
+//The outer loop, now loops over incrememnts of 32 (initially chosen arbitrarliy), and creates tasks of this size
```

Comment:
s/incrememnts/increments/ and s/arbitrarliy/arbitrarily/ and please break this line and the long ones below.

Although this comment and the ones below talk about increments of 32, the code seems to be parametrised to use increments of TASK_SIZE (which is nicer). If that's correct then please could you update the comments. Maybe also say where TASK_SIZE is set.

Reply:
Duh! I see that TASK_SIZE is right there but thanks for adding the comment anyway :-)